documentation/boltspec_reference.md

```diff
@@ -1225,6 +1186,203 @@ The `allow_upload` function accepts the following modifiers:
+  `PlanResult` object returned by this modifier always has a `success` status.
+
+  ```ruby
+  allow_any_plan.always_return(318)
+  ```
+
+- `return(&block) { |plan, params| ... }`
+
+  Invokes a block to construct the `PlanResult` returned by the `run_plan`
+  function.
+
+  ```ruby
+  allow_any_plan.return do |plan, params|
+    Bolt::PlanResult.new(100, 'success')
+  end
+  ```
+
+- `error_with(error)`
+
+  Sets the value for the `PlanResult` object returned by each plan. The
+  `PlanResult` object returned by this modifier always has a `failure` status.
+
+  ```ruby
+  allow_any_plan.error_with('Too many apples, buffer overflow!')
+  ```
+
+### `allow_any_task`
+
+The `allow_any_task` function stubs the [`run_task`
+function](plan_functions.md#run-task). It does not accept parameters.
+
+The `allow_any_task` function accepts the following modifiers:
+
+- `be_called_times(number)`
+
+  The test fails if the any task is run more than _number_ times.
+
+  ```ruby
+  allow_any_task.be_called_times(3)
+  ```
+
+- `not_be_called`
+
+  The test fails if any task is run.
+
+  ```ruby
+  allow_any_task.not_be_called
+  ```
+
+- `with_targets(targets)`
+
+  The target or list of targets that any task can be run on. The test fails if
+  any task is run on a different list of targets.
+
+  ```ruby
+  allow_any_task.with_targets(['target1', 'target2', 'target3'])
+  ```
+
+- `with_params(parameters)`
+
+  The parameters and [options](plan_functions.md#run-script) that can be passed to the
+  `run_task` function. The test fails if any task is run with a different
+  set of parameters and options.
+
+  ```ruby
+  allow__any_task.with_params({ 'breed' => 'german shepherd' })
+  ```
+
+- `always_return(value)`
+
+  Sets the value for each target's `Result` for each task that is run. Returns a
+  `Bolt::ResultSet` object.
+
+  ```ruby
+  allow_any_task.always_return({ 'happy' => true })
+  ```
+
+- `return_for_targets(targets_to_values)`
+
+  Sets the value for each target's `Result` when each task is run. Accepts a
+  hash of key-value pairs where each key is a target and the value is the value
+  for that target's `Result`.
+
+  ```ruby
+  allow_any_task('pet_dog').return_for_targets(
+    'target1' => { 'happy' => true },
+    'target2' => { 'happy' => false }
+  )
+  ```
+
+- `return(&block) { |targets, task, params| ... }`
+
+  Invokes a block to construct the `ResultSet` returned by the `run_task`
+  function.
+
+  ```ruby
+  allow_any_task.return do |targets, task, params|
+    results = targets.map do |target|
+      Bolt::Result.new(target, value: { 'happy' => true })
+    end
+
+    Bolt::ResultSet.new(results)
+  end
+  ```
+
+- `error_with(error)`
+
+  Sets the error hash for each target's `Result` for each task that is run. Returns a
+  `Bolt::ResultSet` object.
+
+  ```ruby
+  allow_any_task.error_with('msg' => 'There are no German Shepherds to pet.')
+  ```
+
+## Execution modes
```

Comment:
Why did this section get moved to the end of the page?

Reply:
Not really sure. 🤔. I'll move it back to the top :)